django/views/generic/edit.py

```diff
@@ -276,12 +279,24 @@ def get_success_url(self):
                 "No URL to redirect to. Provide a success_url.")
 
 
-class BaseDeleteView(DeletionMixin, BaseDetailView):
+class BaseDeleteView(DeletionMixin, FormMixin, SingleObjectMixin, ProcessFormView):
     """
     Base view for deleting an object.
 
     Using this base class requires subclassing to provide a response mixin.
     """
+    form_class = Form
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().get(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super(DeletionMixin, self).post(request, *args, **kwargs)
```

Comment:
1 - Why `BaseDeleteView.delete` calls `post` from `DeletionMixin`? `DeletionMixin.post` already calls `delete`. (It seems to me that it will result in circular method calling.)
2 - Why `super(DeletionMixin, self).post(request, *args, **kwargs)` is being used instead of `super(BaseDeleteView, self).post(request, *args, **kwargs)` ?